src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ImplementIDisposableCorrectly.cs

```diff
@@ -106,13 +114,506 @@ public abstract class ImplementIDisposableCorrectlyAnalyzer : DiagnosticAnalyzer
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: HelpLinkUri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(IDisposableReimplementationRule, FinalizeOverrideRule, DisposeOverrideRule, DisposeSignatureRule, RenameDisposeRule, DisposeBoolSignatureRule, DisposeImplementationRule, FinalizeImplementationRule, ProvideDisposeBoolRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(
+                context =>
+                {
+                    INamedTypeSymbol disposableType = WellKnownTypes.IDisposable(context.Compilation);
+                    if (disposableType == null)
+                    {
+                        return;
+                    }
+
+                    var disposeInterfaceMethod = disposableType.GetMembers(DisposeMethodName).Single() as IMethodSymbol;
+                    if (disposeInterfaceMethod == null)
+                    {
+                        return;
+                    }
+
+                    INamedTypeSymbol garbageCollectorType = context.Compilation.GetTypeByMetadataName(GarbageCollectorTypeName);
+                    if (garbageCollectorType == null)
+                    {
+                        return;
+                    }
+
+                    var suppressFinalizeMethod = garbageCollectorType.GetMembers(SuppressFinalizeMethodName).Single() as IMethodSymbol;
+                    if (suppressFinalizeMethod == null)
+                    {
+                        return;
+                    }
+
+                    var analyzer = new Analyzer(context.Compilation, disposableType, disposeInterfaceMethod, suppressFinalizeMethod);
```

Comment:
Consider: pass `Compilation` to ctor so that ctor returns a fully initialized object and `Initialize` is not needed.